frontend/packages/dev-console/src/components/health-checks/AddHealthCheckWrapper.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as yup from 'yup';
+import { Formik } from 'formik';
+import { FirehoseResult, LoadingBox, StatusBox, history } from '@console/internal/components/utils';
+import { K8sResourceKind, k8sUpdate, modelFor, referenceFor } from '@console/internal/module/k8s';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import { healthChecksProbesValidationSchema } from './health-check-probe-validation-utils';
+import { constructProbeData, getHealthChecksData } from './create-health-check-probe-utils';
+import AddHealthCheck from './AddHealthCheck';
+
+type AddHealthCheckWrapperProps = {
+  resource?: FirehoseResult<K8sResourceKind>;
+  currentContainer: string;
+};
+
+const AddHealthCheckWrapper: React.FC<AddHealthCheckWrapperProps> = ({
+  resource,
+  currentContainer,
+}) => {
+  if (!resource.loaded && _.isEmpty(resource.loadError)) {
+    return <LoadingBox />;
+  }
+
+  if (resource.loadError) {
+    return <StatusBox loaded={resource.loaded} loadError={resource.loadError} />;
+  }
+
+  const container = _.filter(
+    resource.data.spec.template.spec.containers,
+    (data) => data.name === currentContainer,
+  );
+
+  if (_.isEmpty(container)) {
+    return <div className="text-center">Container not found</div>;
+  }
+
+  const handleSubmit = (values, actions) => {
+    const {
+      healthChecks: { readinessProbe, livenessProbe, startupProbe },
+      containerName,
+    } = values;
+    const updatedResource = _.cloneDeep(resource.data);
```

Comment:
Move all this logic into a utility function that takes the resource data and returns the augmented object which has all the health checks added to it.